package proto_ff;

import "DBProxy.proto";
import "DBProxy2.proto";

import "Com.proto";
import "ComDefine.proto";

//中心服通知网关退出游戏  NOTIFY_GATE_LEAVE_GAME
message NotifyGateLeaveGame2
{
  optional uint64 clientid = 1;
  optional LOGOUT_TYPE leave_flag = 3;
}

//网关通知中心服客户端掉线 NOTIFY_CENTER_DISCONNECT
message NotifyCenterDisConnect2
{
  optional uint64 clientId = 1;
  optional uint64 cid =2;
  optional int32	errflag=3;
}

message ProxyServerBusInfo
{
  optional uint32 server_type = 1;
  optional uint32 bus_id = 2;
}

message WroldToProxyChangeServerBusId
{
  repeated ProxyServerBusInfo server_info = 1;
}

message WorldToOtherServerDisconnectNotify
{
  optional uint64 roleId = 1;
  optional uint32 reason = 2;
}

//中心服通知逻辑服退出游戏      NOTIFY_LOGIC_LEAVE_GAME_REQ
message NotifyLogicLeaveGameReq2
{
  required uint64 cid = 1;
  required uint64 uid = 2;
  optional LOGOUT_TYPE  type = 3;
}

//逻辑服返回中心服退出游戏结果   NOTIFY_LOGIC_LEAVE_GAME_RSP
message NotifyLogicLeaveGameRsp2
{
  required uint64 cid = 1;
  required uint64 uid = 2;
  optional int32 logout_type = 3;	//登出类型
}
